File: Problem2.js
function solve(arr){
    if(!arr.length){
      return 0;
    }

    let result = [0, -1, []];

    for(let i = 0; i< arr.length; i++){
      if(arr[i] < 0) continue;

      let prev = (i-1 == 0) ? null: arr[i-1];
      let next = (i+1 == arr.length) ? null : arr[i+1];

      let nxt = [];
      for(let j=0; j<i-1; j++){
        if(arr[j] == prev || arr[j] == next){
          continue;
        }
        nxt.push(arr[j])
      }

      for(let j=i+2; j<arr.length; j++){
        if(arr[j] == prev || arr[j] == next){
          continue;
        }
        nxt.push(arr[j]);
      }

      let maxi = arr[i] + (nxt.length ? Math.max(0, ...nxt) : 0);

      if(result[0] < maxi){
        result = [maxi, i, nxt];
      }else if(result[0] == maxi && arr[i] > arr[result[1]]){
        result = [maxi, i, nxt];
      }
    }
    return result[1] >= 0 ? arr[result[1]] + solve(result[2]) : 0;
}

console.log(solve([1,1,4,1,1]))
console.log(solve([1,1,1,1,1,1,1,10]))
console.log(solve([-1,3,2,3,4,3]))
console.log(solve([-1,-2,-3]))
console.log(solve([0,1,2,3,4,5,6,7,8,9,10]))

File: Problem2.py
def solve(arr):
    if not arr: return 0
    result = (0, -1, [])

    for i in range(len(arr)):
        if arr[i] < 0: continue
        prev = None if i -1 < 0 else arr[i-1]
        next = None if i + 1 == len(arr) else arr[i+1]
        nxt = []
        for j in range(0, i-1):
            if arr[j] == prev or arr[j] == next: continue
            nxt.append(arr[j])

        for j in range(i+2, len(arr)):
            if arr[j] == prev or arr[j] == next: continue
            nxt.append(arr[j])



        maxi = arr[i] + (max(0, max(nxt)) if nxt else 0)

        #print(arr[i], maxi ,nxt)
        if result[0] < maxi:
            result = (maxi, i, nxt)
        elif result[0] == maxi and arr[i] > arr[result[1]]:
            result = (maxi, i, nxt)


    return arr[result[1]] + solve(result[2]) if result[1] >=0 else 0


print(solve([1,1,4,1,1]))
print(solve([1,1,1,1,1,1,1,10]))
print(solve([-1,3,2,3,4,3]))
print(solve([-1,-2,-3]))
print(solve([i for i in range(11)]))
